client/rest/src/plugins/rosetta/nem/OperationParser.js

```diff
@@ -0,0 +1,364 @@
+		return new Transaction(new TransactionIdentifier(metadata.hash), operations);
+	}
+
+	/**
+	 * Parses a transaction into operations.
+	 * @param {object} transaction Transaction to process (REST object model is assumed).
+	 * @returns {Array<Operation>} Transaction operations.
+	 */
+	async parseTransaction(transaction) {
+		const allSignerPublicKeyStringSet = new Set();
+
+		const operations = [];
+		const appendOperation = operation => {
+			// filter out zero transfers
+			if ('transfer' === operation.type && '0' === operation.amount.value)
+				return;
+
+			operation.operation_identifier.index = operations.length;
+			operations.push(operation);
+		};
+
+		const processSubTransaction = async subTransaction => {
+			const subOperations = await this.parseTransactionInternal(subTransaction);
+			subOperations.forEach(appendOperation);
+		};
+
+		if (transaction.otherTrans) {
+			await processSubTransaction(transaction.otherTrans);
+
+			allSignerPublicKeyStringSet.add(transaction.otherTrans.signer);
+			let cosignerFeeSum = 0;
+			transaction.signatures.forEach(signature => {
+				allSignerPublicKeyStringSet.add(signature.signer);
+				cosignerFeeSum += parseInt(signature.fee, 10);
+
+				const cosignOperation = this.createOperation(operations.length, 'cosign');
+				cosignOperation.account = this.publicKeyStringToAccountIdentifier(signature.signer);
+				operations.push(cosignOperation);
+			});
+
+			if (this.options.includeFeeOperation) {
+				const feeCurrency = await this.lookupFeeCurrency();
+				appendOperation(this.createDebitOperation({
+					sourcePublicKey: transaction.otherTrans.signer,
+					amount: parseInt(transaction.fee, 10) + cosignerFeeSum,
+					currency: feeCurrency
+				}));
+			}
+		} else {
+			await processSubTransaction(transaction);
+
+			allSignerPublicKeyStringSet.add(transaction.signer);
+
+			if (this.options.includeFeeOperation) {
+				const feeCurrency = await this.lookupFeeCurrency();
+				appendOperation(this.createDebitOperation({
+					sourcePublicKey: transaction.signer,
+					amount: transaction.fee,
+					currency: feeCurrency
+				}));
+			}
+		}
+
+		return {
+			operations,
+			signerAddresses: [...allSignerPublicKeyStringSet].map(publicKeyString =>
+				this.network.publicKeyToAddress(new PublicKey(publicKeyString)))
+		};
+	}
+
+	/**
+	 * Parses a transaction into operations.
+	 * @param {object} transaction Transaction to process (REST object model is assumed).
+	 * @returns {Array<Operation>} Transaction operations.
+	 * @private
+	 */
+	async parseTransactionInternal(transaction) {
+		const operations = [];
+
+		const pushDebitCreditOperations = (sourcePublicKey, targetAddress, amount, currency) => {
+			operations.push(this.createDebitOperation({ sourcePublicKey, amount, currency }));
+			operations.push(this.createCreditOperation({ targetAddress, amount, currency }));
+		};
+
+		const pushRegistrationFeeOperation = async () => {
+			const amount = transaction.rentalFee;
```

Comment:
Should ``rentalFee`` be ``creationFee`` for mosaics definition tx?
https://nemproject.github.io/#namespaces-and-mosaics

Reply:
yea, you're right. updated.